sdk/core/System.ClientModel/src/Message/PipelineMessage.cs

```diff
@@ -35,7 +33,46 @@ protected internal PipelineMessage(PipelineRequest request)
 
     #region Pipeline invocation options
 
-    public CancellationToken CancellationToken { get; set; }
+    public CancellationToken CancellationToken
+    {
+        get;
+
+        // Set internally by RequestOptions.Apply
+        protected internal set;
+    }
+
+    /// <summary>
+    /// Gets or sets the message classifier used by the client pipeline to
+    /// determine whether the service responded with an error response, and
```

Comment:
nit: I think we should make it slightly clearer that this specifically affects how the response will be interpreted rather than changing anything about the response itself. 

```suggestion
    /// determine whether the response from the service should be considered an error, and
```